### PR TITLE
added Signup with email and password

### PR DESCRIPTION
**Pull request by YashKamboj**
**Issue Number**
#396 
fixes #

**Describe the changes you've made**

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

**Additional context (OPTIONAL)**

**Test plan (OPTIONAL)**

A good test plan should give instructions that someone else can easily follow.

**Checklist**

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**
Preview: Deploy preview link here with the appropriate route

<!-- Please add deployed link to here text >
@Abhishek-kumar09

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
https://user-images.githubusercontent.com/48255244/132378844-04d20689-4db7-49da-99a5-ce6d49c3c5a9.mp4

**Review by Abhishek-kumar09 — COMMENTED** [review]
Awesome @YashKamboj ,
Few changes for future:
- Add email validation on textField.
- Add obscure password field
- Add a progress bar that shows the login is under progress

**Comment by YashKamboj** [comment]
Thank you .
Do i make the new features now??
@Abhishek-kumar09

**Comment by Abhishek-kumar09** [comment]
@YashKamboj You can keep it in different PR or same, your choice :smile:

File: src/components/AuthComponent/SignupSection.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
 import { makeStyles } from "@material-ui/core/styles";
 import { useDispatch } from "react-redux";
-import { loginAction } from "../../services/authService";
+import { loginAction, loginActionWithEmailPassword } from "../../services/authService";
 
 function SignupSection() {
   const dispatch = useDispatch();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [displayName, setDisplayName] = useState("");
```
Comment (Abhishek-kumar09) [inline src/components/AuthComponent/SignupSection.js:11]: combine these three in one Object like:
```
const [user, setUser] = useState({});
```
 to change the particular key in setState,
 ```
 setUser({...user, key: value})
 ```

File: src/services/authService.js [hunk 2]
```diff
@@ -1,71 +1,74 @@
...
-      });
-    }
-   
-    
+export const loginAction = () => {
+  return (dispatch) => {
+    dispatch(login());
+    cfaSignIn("google.com").subscribe((user) => {
+      setuserData(user.displayName, user.email, user.photoURL);
+      user.sendEmailVerification();
```
Comment (Abhishek-kumar09) [inline src/services/authService.js:39]: I think it's google signIn, this is mixing google signIn with email and password auth

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: src/components/AuthComponent/SignupSection.js (hunk 1 above)
Comment (YashKamboj) [reply src/components/AuthComponent/SignupSection.js:11]: Sorry! I am unable to combine ,when i combine then error shows of not defined , can you teach me how to do it.

Comment (Abhishek-kumar09) [reply src/components/AuthComponent/SignupSection.js:11]: Please take a look at how it done.

https://github.com/codeforcauseorg/codeforcause.org/blob/ff1261f05941b4e5618f9ff7890a76e4ac1f91aa/src/views/pages/CLView/ApplyNowModal.js#L47-L67

The formData is the combined that that contains all the information about the form. See how it is used in textFields

File: src/services/authService.js (hunk 2 above)
Comment (YashKamboj) [reply src/services/authService.js:39]: Hey i am really sorry , i do not know the user comes empty , can you please help

Comment (YashKamboj) [reply src/services/authService.js:39]: i am really stuck

Comment (adarsh-technocrat) [reply src/services/authService.js:39]: Hey, Yash let's join on CFC discord under the project hangout section at 6 pm tomorrow I will explain the complete code base if it's fine.  

cc/ @Abhijay007  @yashvardhan05

Comment (adarsh-technocrat) [reply src/services/authService.js:39]: and we will discuss this issue also :)

Comment (YashKamboj) [reply src/services/authService.js:39]: Yes , ok thank you

**Comment by yashvardhan05** [comment]
Yes!! hoping there at 6 :)

**Comment by YashKamboj** [comment]
@Abhijay007 any progress

**Comment by Abhijay007** [comment]
> @Abhijay007 any progress

@YashKamboj sorry for the late reply buddy, I tried it but was not able to find any optimal solutions, I suggest you ask @adarsh-technocrat about this issue as he already implemented something like this, you can message him on discord, maybe he can help you with this !!

**Comment by YashKamboj** [comment]
> > @Abhijay007 any progress
> 
> @YashKamboj sorry for the late reply buddy, I tried it but was not able to find any optimal solutions, I suggest you ask @adarsh-technocrat about this issue as he already implemented something like this, you can message him on discord, maybe he can help you with this !!

Ok thnx for your time

**Pull request closed without merging**